cfgmgr/buffermgrdyn.cpp

```diff
@@ -309,6 +309,9 @@ void BufferMgrDynamic::calculateHeadroomSize(buffer_profile_t &headroom)
             if (pairs[0] == "xon_offset")
                 headroom.xon_offset = pairs[1];
         }
+
```

Comment:
should you check empty before the for loop or this must be after the for loop?
i think as an error for the user you should refer to what is empty. the empty concept is internal to the code IMO

Reply:
Yes. We should check empty before the loop.
As the Lua script doesn't return anything, what we can tell the user is that the script failed to execute.
From the code, in any case, this should not happen. We just add the message just in case.